docs/DeveloperGuide.md

```diff
@@ -1,38 +1,254 @@
+the `BudgetAction` class.
+The class diagram below shows how the `BudgetCommand` parent class is implemented, as well as its extended classes.
+
+![BudgetCommand.png](uml/BudgetCommand.png)
+
+#### BudgetCommand Sequence
+
+Once a string `input` from the user has been deemed as a `Budget` command as explained in the `CommandParser` sequence
+diagram above,
+the `input` will be passed into the `BudgetCommand` class. This class determines what budget commands to carry out to
+update the `data` stored in the `Budget` and `Expense` array lists, as shown in the sequence diagram below.
+
+![BudgetCommandSequence.png](uml/BudgetCommandSequence.png)
+
+### Deposit Component
 
-### Value proposition
+### DepositCommand Class
 
-{Describe the value proposition: what problem does it solve?}
+The `DepositCommand` class contains methods that relate to the execution of the deposit functionality in BudgetBuddy.
+Users can
+create new deposits with a timestamp (if necessary). These deposits are stored in a deposit list and users can delete
+deposits, find
+deposits using keywords, and list all deposits.
 
-## User Stories
+The user's input is split by the parser in the `CommandParser` class and is redirected to the 'DepositCommand' class if
+the first word
+is "deposit." Based on the second word, a method in `DepositAction` class is called corresponding to the command
+requested by the user.
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+Attached below is how the `DepositCommand` class is implemented along with its relation with the other `Data` classes
+and the abstract
+`Command` class.
 
-## Non-Functional Requirements
+![DepositCommand.png](uml/DepositCommand.png)
 
-{Give non-functional requirements}
+### Design & Implementation of the Deposit Feature
 
-## Glossary
+Like all other functionalities of BudgetBuddy, the deposit feature is heavily modularized and designed with an OOP lens.
+Because of this,
+the parsing of user input, the parsers for each feature, and each feature's actions are all in separate classes.
 
-* *glossary item* - Definition
+On a high level, the deposit feature starts with `CommandParser` taking in the input and choosing which `Command` class
+to execute from.
+This happens with all user input in BudgetBuddy. Then if the first word is "deposit," the `execute` function of
+the `DepositCommand` class
+will run, creating a new `DepositAction` class. The `execute` function will run a method corresponding to what the user
+inputs. The design
+of this three class system is meant to modularize the different aspects of the internal logic so future problems would
+be encapsulated in a specific location.
 
-## Instructions for manual testing
+### Expense Component
```

Comment:
as mentioned for the depositCommand section, likewise for expense and stats command class